apps/server/src/shared/repo/user/user.repo.integration.spec.ts

```diff
@@ -413,43 +413,65 @@ describe('user repo', () => {
 		});
 	});
 
-	describe('delete', () => {
-		const setup = async () => {
-			const user1: User = userFactory.buildWithId();
-			const user2: User = userFactory.buildWithId();
-			const user3: User = userFactory.buildWithId();
-			await em.persistAndFlush([user1, user2, user3]);
+	describe('deleteUser', () => {
+		describe('when user does not exist', () => {
+			const setup = () => {
+				const user = userFactory.buildWithId();
 
-			return {
-				user1,
-				user2,
-				user3,
+				return {
+					user,
+				};
 			};
-		};
-		it('should delete user', async () => {
-			const { user1, user2, user3 } = await setup();
-			const deleteResult = await repo.deleteUser(user1.id);
-			expect(deleteResult).toEqual(1);
-
-			const result1 = await em.find(User, { id: user1.id });
-			expect(result1).toHaveLength(0);
-
-			const result2 = await repo.findById(user2.id);
-			expect(result2).toMatchObject({
-				firstName: user2.firstName,
-				lastName: user2.lastName,
-				email: user2.email,
-				roles: user2.roles,
-				school: user2.school,
+
+			it('should return empty array', async () => {
+				const { user } = setup();
+
+				const result = await repo.deleteUser(user.id);
+
+				expect(result).toBeNull();
 			});
+		});
+		describe('when user exists', () => {
+			const setup = async () => {
+				const user1: User = userFactory.buildWithId();
+				const user2: User = userFactory.buildWithId();
+				const user3: User = userFactory.buildWithId();
+				await em.persistAndFlush([user1, user2, user3]);
 
-			const result3 = await repo.findById(user3.id);
-			expect(result3).toMatchObject({
-				firstName: user3.firstName,
-				lastName: user3.lastName,
-				email: user3.email,
-				roles: user3.roles,
-				school: user3.school,
+				const expectedResult = user1.id;
+
+				return {
+					expectedResult,
+					user1,
+					user2,
+					user3,
+				};
+			};
+			it('should delete user', async () => {
+				const { expectedResult, user1, user2, user3 } = await setup();
+				const deleteResult = await repo.deleteUser(user1.id);
+				expect(deleteResult).toEqual(expectedResult);
+
+				const result1 = await em.find(User, { id: user1.id });
+				expect(result1).toHaveLength(0);
+
+				const result2 = await repo.findById(user2.id);
+				expect(result2).toMatchObject({
+					firstName: user2.firstName,
+					lastName: user2.lastName,
+					email: user2.email,
+					roles: user2.roles,
+					school: user2.school,
+				});
+
+				const result3 = await repo.findById(user3.id);
+				expect(result3).toMatchObject({
+					firstName: user3.firstName,
+					lastName: user3.lastName,
+					email: user3.email,
+					roles: user3.roles,
+					school: user3.school,
+				});
```

Comment:
This function feels a bit too long, could you please move these props objects to the `setup()` function? And also please think about a little more descriptive name for this test case, e.g. "should delete given user, but not the other ones" as I think that's exactly what you're checking within this test.

Reply:
done